### ISSUE TITLE
tf.keras.layers.BatchNormalization returns output that's significantly different from batch norm formula

### ISSUE THREAD

**Issue by funkyyyyyy** [keras-team/keras#15592]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.6.0
- Python version: 3.7
- Bazel version (if compiling from source):
- GPU model and memory: RTX 3060 6GB
- Exact command to reproduce: Run the colab notebook

**Describe the problem**.

Given the same `input`, `gamma`, `beta`, `moving_mean`, `moving_var` and `epsilon`, `tf.keras.layers.BatchNormalization` returns an output that's different from expected value (using `numpy`). The difference is in the range of `e^-3`

**Describe the current behavior**.

Difference in output from `tf.keras.layers.BatchNormalization`  vs manual calculation using `numpy` is in the range of `e^-3`

**Describe the expected behavior**.

Difference in output from `tf.keras.layers.BatchNormalization`  vs manual calculation using `numpy` should be in the range of `e^-5` or smaller

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): no

**Standalone code to reproduce the issue**.

Colab notebook to reproduce issue provided at https://colab.research.google.com/drive/1p1LfiTp9_liAj7vtWEllYhQcwxsR_pxr?usp=sharing

**Comment by funkyyyyyy** [comment]
If its any help, I dug deeper into the Keras code and found the source of discrepancy

if the input shape is `(N, C)` or `(N, T, C)`, then `tf.nn.batch_normalization` is used within `tf.keras.layers.BatchNormalization` to calculate the batch normalization. The output value here is in line with numpy outputs. 

However, if the input shape is longer such as `(N, H, W, C) ` or `(N, T, H, W, C)`, then `tf.compat.v1.nn.fused_batch_norm` is used and the large discrepancy in output occurs

**Comment by funkyyyyyy** [comment]
https://github.com/keras-team/keras/issues/15571 might be related to this issue

**Comment by funkyyyyyy** [comment]
this issue can be temporarily fixed by `tf.keras.layers.BatchNormalization(<other args>, fused=False)`

**Comment by chunduriv** [comment]
@jvishnuvardhan, I was able to replicate issues in `TF2.7` and `tf-nightly`. Please find the [gist here](https://colab.research.google.com/gist/chunduriv/98496019a2e16e5ab45dd03636c8ea61/15592.ipynb). Thanks!

**Comment by mattdangerw** [comment]
Took a look, it does look like this difference is directly from `tf.compat.v1.nn.fused_batch_norm`, which won't guarantee the exact same computation. If the difference is important, I think the fix you listed (passing `fused=False`) is the correct for now.

The long term fix we'd like is to move a lot of the complexity out of batch norm, use XLA compilation (`jit_compile=True`) to keep things efficient, and remove the fused op entirely. Until we do that, we will keep the fused option default to avoid a performance regression.

**Comment by tilakrayal** [comment]
@funkyyyyyy,
I tried to execute the mentioned code on tf-nightly(2.14.0-dev20230508) and tried with tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=0.00001, name='tf_batch_norm', fused=False) & 
tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=0.00001, name='tf_batch_norm') , got the difference as  **2.2888184e-05** which was different from the older version. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/46a9b25f0c657627783242b6a3f5ef27/batch_normalization_output_comparison.ipynb). Thank you!
